### PR TITLE
GG-5074

### PR DESCRIPTION
**Pull request by martin-armstrong**

### CODE REVIEW

File: app/uk/gov/hmrc/emailverification/services/AuditService.scala [hunk 1]
```diff
@@ -30,38 +31,193 @@ class AuditService @Inject() (
...
     val details = Map(
-      "pinCode" -> passcode,
-      "emailVerified" -> "true",
+      "emailAddress" -> emailAddress,
+      "emailVerified" -> failureReason.fold("true")(_ => "false"),
+      "responseCode" -> responseCode.toString,
+      "bearerToken" -> hc.authorization.getOrElse(Authorization("-")).value
+    ) ++ failureReason.fold[Map[String, String]](Map()) { reason =>
+        Map("verifyFailureReason" -> reason)
+      }
```
Comment (richard-dennehy) [inline app/uk/gov/hmrc/emailverification/services/AuditService.scala]: if you like abusing the scala compiler
```suggestion
    ) ++ failureReason.map("verifyFailureReason" -> _).toMap
``` 
works

File: app/uk/gov/hmrc/emailverification/services/AuditService.scala [hunk 2]
```diff
@@ -30,38 +31,193 @@ class AuditService @Inject() (
     auditConnector: AuditConnector
 )(implicit ec: ExecutionContext) extends BackendHeaderCarrierProvider {
 
-  def sendPinViaEmailEvent(emailAddress: String, passcode: String, serviceName: String, responseCode: Int)(implicit request: Request[_]) = {
+  def sendPasscodeViaEmailEvent(emailAddress: String, passcode: String, serviceName: String, responseCode: Int)(implicit request: Request[_]) = {
```
Comment (richard-dennehy) [inline app/uk/gov/hmrc/emailverification/services/AuditService.scala:39]: we need unit tests for this lot to ensure that e.g. the tags are correct, as per our agreement with TXM/whatever they call themselves these days

**Review by richard-dennehy — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/emailverification/services/AuditService.scala (hunk 1 above)
Comment (martin-armstrong) [reply app/uk/gov/hmrc/emailverification/services/AuditService.scala]: 👍

File: app/uk/gov/hmrc/emailverification/services/AuditService.scala (hunk 2 above)
Comment (martin-armstrong) [reply app/uk/gov/hmrc/emailverification/services/AuditService.scala:39]: 👍

**Review by richard-dennehy — APPROVED** [review]

**Pull request merged**